AK/InsertionSort.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2022, Marc Luqué <marc.luque@outlook.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/StdLibExtras.h>
+
+namespace AK {
+
+/* Standard Insertion Sort.
+ * NOTE: `end` is inclusive.
+ */
+template<typename Collection, typename Compare>
```

Comment:
Compare -> Comparator. You might want to use a `Function`-based concept to facilitate better errors.

Reply:
That's a really cool idea! Thank you for pointing that out. I searched a bit in the codebase and found [Concepts.h](https://github.com/SerenityOS/serenity/blob/master/AK/Concepts.h) which has `VoidFunction`. I think that I could make some modified version of that to account for something like `(T a, U b) -> bool`. Should we also engage in the "comparability" of T and U? There seems to be some [pre-defined concept](https://en.cppreference.com/w/cpp/concepts/totally_ordered) for that. I think that this would fit quite well since `sort => total order`, i.e., a total order is a necessary condition for sorting.

Reply:
In my other comment I said that you need `Function<bool(T, T)>` IIRC

Reply:
Ah I see now, there's already a concept for it, I didn't see that 🤦🏼‍♂️

What do you think about enforcing a total order on `T`? Thinking about it again, I think that it could break quite some stuff since it would require `T` to define all the necessary operators `<`, `<=`, `>`, `>=`, `==`, `!=`. On the other hand side, it's a really neat thing to make sure at compile time that >= or < is an order on `T` that results in a correct sorting.

Reply:
I think that doesn't make sense, as the comparator function can do whatever it wants. If you don't use the default constructed function, T doesn't need to be totally ordered. If you do use it, you'll get a rather good error message right away.

Reply:
I see, in that case, I'll leave it and mark this as resolved ^^